### PR TITLE
Change URLs and naming from logtail.com to logs.betterstack.com

### PR DESCRIPTION
**Pull request by PetrHeinz**
Changing links and copy after the rename Logtail -> Better Stack Logs

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 description: |-
-  The Logtail provider.
+  The Better Stack Logs provider.
 ---
 
-# Logtail Provider
+# Better Stack Logs
 
-The [Logtail](https://logtail.com) provider provides resources to interact with the [Logtail API](https://docs.logtail.com/api/getting-started).
+[Better Stack Logs](https://logs.betterstack.com) provider provides resources to interact with its [API](https://betterstack.com/docs/logs/api/getting-started/).
```
Comment (PetrHeinz) [inline docs/index.md]:
```suggestion
[Better Stack Logs](https://logs.betterstack.com) provider provides resources to interact with the [Logs API](https://betterstack.com/docs/logs/api/getting-started/).
```

File: templates/index.md.tmpl [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 description: |-
-  The Logtail provider.
+  The Better Stack Logs provider.
 ---
 
-# Logtail Provider
+# Better Stack Logs
 
-The [Logtail](https://logtail.com) provider provides resources to interact with the [Logtail API](https://docs.logtail.com/api/getting-started).
+[Better Stack Logs](https://logs.betterstack.com) provider provides resources to interact with its [API](https://betterstack.com/docs/logs/api/getting-started/).
```
Comment (PetrHeinz) [inline templates/index.md.tmpl]:
```suggestion
[Better Stack Logs](https://logs.betterstack.com) provider provides resources to interact with the [Logs API](https://betterstack.com/docs/logs/api/getting-started/).
```

File: examples/basic/variables.tf [hunk 3]
```diff
@@ -2,7 +2,7 @@ variable "logtail_api_token" {
   type        = string
   description = <<EOF
 Logtail API Token
```
Comment (curusarn) [inline examples/basic/variables.tf:4]: Better Stack Logs

File: templates/index.md.tmpl [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
 layout: ""
-page_title: "Provider: Logtail"
+page_title: "Provider: logtail"
 description: |-
-  The Logtail provider.
+  The Better Stack Logs provider.
 ---
 
-# Logtail Provider
+# Better Stack Logs
```
Comment (curusarn) [inline templates/index.md.tmpl:8]: Provider

**Review by curusarn — COMMENTED** [review]
Looks good. Thank you! Two nitpicks.

**Pull request merged**